### ISSUE TITLE
It does not work on Microsoft Windows

### ISSUE THREAD

**Issue by HenriMorlaye** [appsignal/appsignal-elixir#36]
I guess it's not a priority, just to make sure it's recorded somewhere.

** (FunctionClauseError) no function clause matching in Mix.Appsignal.Helper.map_arch/1
    mix_helpers.exs:103: Mix.Appsignal.Helper.map_arch('win32')

**Comment by arjan** [comment]
@thijsc is windows supported at all for Appsignal?

**Comment by thijsc** [comment]
No, and we're not planning Windows support. The integration should work without breaking anything though. In Ruby there are empty implementations of every method if the extension is not loaded.

**Comment by tombruijn** [comment]
As mentioned before and in our documentation (http://docs.appsignal.com/support/operating-systems.html#microsoft-windows) we currently have no plans to support Windows because we don't hear a lot from people that use it. But let us know when you do use it so we know if there's a lot of demand for it.

Making sure you can run AppSignal on a non-supported architecture without it crashing the app will be handled in #89 

Closing

**Comment by HenriMorlaye** [comment]
Hi, it does not work on Bash For Windows either (Ubuntu 16.04 shipped with Windows 10 Creators Update). Elixir and Phoenix work fine though.
The error: 

[error] Error loading NIF, AppSignal integration disabled!
Failed to load NIF library: '/mnt/c/blablabla/_build/dev/lib/appsignal/priv/appsignal_extension.so: cannot enable executable stack as shared object requires: Invalid argument'
Is your operating system (x86_64-unknown-linux-gnu) supported?

**Comment by arjan** [comment]
Bash for Windows is still windows, under the hood, I think.

**Comment by HenriMorlaye** [comment]
Yes, very deep down, but it basically enables native Linux ELF64 binaries to run on Windows without any VM. Both Erlang's and Elixir's ubuntu packages do build and work on it. Bash On Windows is a major argument for linux or mac users switching to Windows 10. They keep developping on an almost normal Ubuntu subsystem without running a VM.

**Comment by spapas** [comment]
This is an old issue however I'd like to know if app signal spports Window 10 now. And if it still doesn't support Window I'ld like to know why is that?

**Comment by shairyar** [comment]
We don't support it because there's not a lot of demand for it that we've seen. Most people won't use windows in their production apps, are you using windows on your production server?

You can give the [windows subsystem for Linux](https://docs.appsignal.com/support/operating-systems.html#microsoft-windows-subsystem-for-linux) a try.

**Comment by spapas** [comment]
Hello @shairyar thanks for replying.

No of course we don't use windows on production. However I'm using windows as my dev system and I'd really like to be able to debug the appsignal integration with my app. I haven't tested it in WSL yet because comment https://github.com/appsignal/appsignal-elixir/issues/36#issuecomment-293594304 says that it doesn't work, do you think that something may have changed? (I don't use WSL normally)

TIA

**Comment by shairyar** [comment]
That comment is from 2017, I am not sure if it's still broken but we will take a look and see. If it is broken we will fix it.

**Comment by tombruijn** [comment]
Let's test the Elixir integration on WSL on a virtual windows machine. If anything breaks, let's fix it. Do remember to install the correct build packages for linux in WSL. https://docs.appsignal.com/support/operating-systems.html
